File: Activities/app/src/main/java/com/example/activities/MainActivity.java
package com.example.activities;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {


    public static final String EXTRA_MESSAGE = "com.example.android.activities.extra.MESSAGE";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
}
    public void launchSecondActivity(View view){
        Intent openSecondActivity = new Intent(this,SecondActivity.class);
        String passage = "Chill and Display";
        openSecondActivity.putExtra(EXTRA_MESSAGE, passage);
        startActivity(openSecondActivity);

    }
}

File: DiceRoller/app/src/main/java/com/example/diceroller/MainActivity.kt
package com.example.diceroller

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.Button
import android.widget.ImageView
import androidx.databinding.DataBindingUtil
import com.example.diceroller.databinding.ActivityMainBinding
import java.util.*

class MainActivity : AppCompatActivity() {
    lateinit var diceImage: ImageView
    private lateinit var binding: ActivityMainBinding
    private val myName: MyName = MyName()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        binding.myName = MyName()

        binding.rollButton.setOnClickListener{
            addNickName(it)
        }
    }

    private fun addNickName(it: View?) {
        binding.apply {
            myName?.nickname = editTextTextPersonName.text.toString()
            textView.text = editTextTextPersonName.text
        }
        val imm = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        if (it != null) {
            imm.hideSoftInputFromWindow(it.windowToken, 0)
        }
    }

//    private fun rollDice(it: View) {
//        binding.apply {
//        val randomInt = Random().nextInt(6) + 1
//        val drawableResource = when(randomInt){
//                1 -> R.drawable.dice_1
//                2 -> R.drawable.dice_2
//                3 -> R.drawable.dice_3
//                4 -> R.drawable.dice_4
//                5 -> R.drawable.dice_5
//            else ->  R.drawable.dice_6
//        }
//        imageView.setImageResource(drawableResource)
//    }}
//}
 }


